R/content.R

```diff
@@ -925,3 +975,24 @@ get_content_permissions <- function(content, add_owner = TRUE) {
   res <- content$permissions(add_owner = add_owner)
   parse_connectapi_typed(res, connectapi_ptypes$permissions)
 }
+
+
+#' @rdname refresh
+#' @export
+content_refresh <- function(content, variant_key = NULL) {
+  warn_experimental("content_refresh")
+  scoped_experimental_silence()
+  validate_R6_class(content, "Content")
+
+  # TODO: Get all variants, warn if more than one and no key provided.
+  if (!is.null(variant_key)) {
+    variant <- get_variant(content, variant_key)
+  } else {
+    variant <- get_variant_default(content)
+  }
+
+  rendered <- variant$render()
+  rendered$task_id <- rendered$id
+
+  VariantTask$new(connect = variant$get_connect(), content = content$get_content(), key = variant$key, task = rendered)
+}
```

Comment:
This only works for rendered content currently, yeah? You're thinking of swapping out some / most of this with the `$refresh()` version up above, or something else?

Reply:
Yeah, you're right — this was my first pass at using the variant object to do the render, and I took this and updated it for the method implementations above.

We talked a little this week about whether it's better to use methods or functions.

If the methods remain, I'll update this to just call the content item's `$refresh()` method.